torch/nn/functional.py

```diff
@@ -5261,12 +5261,13 @@ def multi_head_attention_forward(
         target_type=query.dtype
     )
 
-    if is_causal and attn_mask is None:
-        raise RuntimeError(
-            "Need attn_mask if specifying the is_causal hint. "
-            "You may use the Transformer module method "
-            "`generate_square_subsequent_mask` to create this mask."
-        )
+    if is_causal:
+        if attn_mask is None:
```

Comment:
Just out of interest, is there a specific reason (e.g. TorchScript) to separate the if-query?

Reply:
There used to be an else-branch on the second if

Reply:
In that case I think it would make sense to revert the change, it's a bit nicer to look at with the combined if-query in my opinion.